registry/coder-labs/modules/gemini/main.tf

```diff
@@ -0,0 +1,215 @@
+  default     = ""
+}
+
+variable "pre_install_script" {
+  type        = string
+  description = "Custom script to run before installing Gemini."
+  default     = null
+}
+
+variable "post_install_script" {
+  type        = string
+  description = "Custom script to run after installing Gemini."
+  default     = null
+}
+
+data "coder_parameter" "ai_prompt" {
+  type        = "string"
+  name        = "AI Prompt"
+  default     = ""
+  description = "Initial prompt for the Gemini CLI"
+  mutable     = true
+}
+
+variable "additional_extensions" {
+  type        = string
+  description = "Additional extensions configuration in json format to append to the config."
+  default     = null
+}
+
+variable "gemini_system_prompt" {
+  type        = string
+  description = "System prompt for Gemini. It will be added to GEMINI.md in the specified folder."
+  default     = ""
+}
+
+resource "coder_env" "gemini_api_key" {
+  agent_id = var.agent_id
+  name     = "GEMINI_API_KEY"
+  value    = var.gemini_api_key
+}
+
+resource "coder_env" "gemini_use_vertex_ai" {
+  agent_id = var.agent_id
+  name     = "GOOGLE_GENAI_USE_VERTEXAI"
+  value    = var.use_vertexai
+}
+
+locals {
+  base_extensions = <<-EOT
+{
+  "coder": {
+    "args": [
+      "exp",
+      "mcp",
+      "server"
+    ],
+    "command": "coder",
+    "description": "Report ALL tasks and statuses (in progress, done, failed) you are working on.",
+    "enabled": true,
+    "env": {
+      "CODER_MCP_APP_STATUS_SLUG": "${local.app_slug}",
+      "CODER_MCP_AI_AGENTAPI_URL": "http://localhost:3284"
+    },
+    "name": "Coder",
+    "timeout": 3000,
+    "type": "stdio",
+    "trust": true
+  }
+}
+EOT
+
+  app_slug        = "gemini"
+  install_script  = file("${path.module}/scripts/install.sh")
+  start_script    = file("${path.module}/scripts/start.sh")
+  module_dir_name = ".gemini-module"
+}
+
+module "agentapi" {
+  source  = "registry.coder.com/coder/agentapi/coder"
+  version = "1.0.0"
+
+  agent_id             = var.agent_id
+  web_app_slug         = local.app_slug
+  web_app_order        = var.order
+  web_app_group        = var.group
+  web_app_icon         = var.icon
+  web_app_display_name = "Gemini"
+  cli_app_slug         = "${local.app_slug}-cli"
+  cli_app_display_name = "Gemini CLI"
+  module_dir_name      = local.module_dir_name
+  install_agentapi     = var.install_agentapi
+  agentapi_version     = var.agentapi_version
+  pre_install_script   = var.pre_install_script
+  post_install_script  = var.post_install_script
+  start_script         = <<-EOT
+     #!/bin/bash
+     set -o errexit
+     set -o pipefail
+
+     echo -n '${base64encode(local.start_script)}' | base64 -d > /tmp/start.sh
+     chmod +x /tmp/start.sh
+     GEMINI_API_KEY='${var.gemini_api_key}' \
+     GOOGLE_GENAI_USE_VERTEXAI='${var.use_vertexai}' \
+     GEMINI_MODEL='${var.gemini_model}' \
+     GEMINI_START_DIRECTORY='${var.folder}' \
+     GEMINI_TASK_PROMPT='${data.coder_parameter.ai_prompt.value}' \
```

Comment:
this should also be escaped with base64 - a multi-line string would break the script